crates/wast-util/src/lib.rs

```diff
@@ -229,18 +229,52 @@ pub struct WastConfig {
     pub collector: Collector,
 }
 
+/// Different compilers that can be tested in Wasmtime.
 #[derive(PartialEq, Debug, Copy, Clone)]
 pub enum Compiler {
+    /// Cranelift backend.
+    ///
+    /// This tests the Cranelift code generator for native platforms. This
+    /// notably excludes Pulley since that's listed separately below even though
+    /// Pulley is a backend of Cranelift. This is only used for native code
+    /// generation such as x86_64.
     Cranelift,
+
+    /// Winch backend.
+    ///
+    /// This tests the Winch backend for native platforms. Currently Winch
+    /// primarily supports x86_64.
     Winch,
+
+    /// Pulley interpreter.
+    ///
+    /// This tests the Cranelift pulley backend plus the pulley execution
+    /// environment of the output bytecode. Note that this is separate from
+    /// `Cranelift` above to be able to test both on platforms where Cranelift
+    /// has native codegen support.
+    Pulley,
```

Comment:
Possible bikeshed: Maybe rename `Cranelift` to `CraneliftNative` and `Pulley` to `CraneliftPulley`? In addition to clarifying the native code execution bits, this would future proof for a possible `WinchPulley` in the future.